jvb/src/main/java/org/jitsi/videobridge/shim/ColibriUtil.java

```diff
@@ -146,7 +149,33 @@ static String getOctoChannelId(MediaType mediaType)
             // video channel is signaled.
             if (MediaType.VIDEO.equals(contentShim.getMediaType()) && !channelSources.isEmpty())
             {
-                channelShim.getEndpoint().recreateMediaSources();
+                List<ChannelShim> videoChannels = contentShim.getChannelShims().stream()
+                        .filter(c -> c.getMediaType() == MediaType.VIDEO)
+                        .filter(c -> c.getEndpoint() == channelShim.getEndpoint())
```

Comment:
I'm a little concerned (for performance reasons) that this is iterating over all the channels in the conference.  Since you added `EndpointShim`, would it make sense to have `recreateMediaSources` reference the `ChannelShim`s from there?